lib/youtube/oauth.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Youtube.OAuth do
+  use Tesla
+
+  @behaviour OAuth
+  @base_url "https://accounts.google.com/o/oauth2/v2/auth"
+  @client_id Application.compile_env!(:roule_avec_spot, [Youtube, :client_id])
+  @client_secret Application.compile_env!(:roule_avec_spot, [Youtube, :client_secret])
+  @redirect_uri Application.compile_env!(:roule_avec_spot, [Youtube, :redirect_uri])
+
+  def get_token() do
+    token = read_token()
+
+    cond do
+      token ->
+        token
+
+      true ->
+        token = OAuth.create_access_token!(Youtube.OAuth)
+        Application.put_env(:roule_avec_spot, :google_token, token)
```

Comment:
I think this case is borderline OK because it is configuration, but I would maybe consider other options to store state, e.g. an Agent or ETS. Typically, for a token, you might need to have to handle expiration as well and re-issue once expired.